package-parser/package_parser/commands/generate_annotations/generate_annotations.py

```diff
@@ -390,3 +393,64 @@ def __is_required(values: list[tuple[str, int]]) -> bool:
         values, key=lambda tup: tup[1]
     )[-2:]
     return most_used_value_tupel[1] - seconds_most_used_value_tupel[1] <= m / n
+
+
+def __get_boundary_annotations(
+    usages: UsageStore, api: API, annotations: AnnotationStore
+) -> None:
+    """
+    Annotates all parameters which are a boundary.
+    :param usages: Usage store
+    :param api: Description of the API
+    :param annotations: AnnotationStore, that holds all annotations
+    """
+    for _, parameter in api.parameters().items():
+        refined_type = parameter.refined_type.as_dict()
+        if "kind" in refined_type and refined_type["kind"] == "BoundaryType":
+            target = __qname_to_target_name(api, parameter.qname)
+            min_value = refined_type["min"]
+            max_value = refined_type["max"]
+
+            is_discrete = __is_discrete(min_value, max_value)
+
+            min_limit_type = 0
+            max_limit_type = 0
+            if not refined_type["min_inclusive"]:
+                min_limit_type = 1
+            if not refined_type["max_inclusive"]:
+                max_limit_type = 1
+            if min_value == "NegativeInfinity":
+                min_limit_type = 2
+            if max_value == "Infinity":
+                max_limit_type = 2
+
+            interval = Interval(
+                isDiscrete=is_discrete,
+                lowerIntervalLimit=min_value,
+                upperIntervalLimit=max_value,
+                lowerLimitType=min_limit_type,
+                upperLimitType=max_limit_type,
+            )
+            boundary = BoundaryAnnotation(
+                target=target,
+                interval=interval,
+            )
+            annotations.boundaries.append(boundary)
+
+
+def __is_discrete(
+    min_value: Union[str, int, float], max_value: Union[str, int, float]
+) -> bool:
+    """
+    Checks whether an interval is discrete or not.
+    :param min_value: Union[str, int, float]
+    :param max_value: Union[str, int, float]
+    :return discretionValue
+    """
+
+    if min_value == "NegativeInfintiy":
+        min_value = 0
+    if max_value == "Infinity":
+        max_value = 0
```

Comment:
Argumente werden in Python "by value" übergeben. Die Änderungen an `min_value` und `max_value` sind außerhalb der Funktion nicht sichtbar.